llama-index-integrations/llms/llama-index-llms-huggingface/llama_index/llms/huggingface/base.py

```diff
@@ -685,6 +686,10 @@ async def astream_complete(
         raise NotImplementedError
 
 
+@deprecated(
```

Comment:
I can't comment that far above, but above this: `class HuggingFaceInferenceAPI(CustomLLM):` should be deprecated and moved to a new package

So this PR would create 4 packages actually
- huggingface utils
- api embeddings
- api llm
- tgi llm